code/_onclick/hud/screen_objects.dm

```diff
@@ -45,8 +45,10 @@
 
 /atom/movable/screen/Initialize(mapload, datum/hud/hud_owner)
 	. = ..()
-	if(hud_owner && istype(hud_owner))
-		hud = hud_owner
+	if(!istype(hud_owner))
+		return
+	hud = hud_owner
+	RegisterSignal(hud, COMSIG_QDELETING, PROC_REF(on_hud_delete))
```

Comment:
this potentially breaks parallax.
Rather then doing something like this, create a setter proc that unreg's/registers QDELETING and such, and replace all other instances of modifying hud with it

Also, this is odd and is I think a patch fix.
Screen objects are meant to be owned by their hud, like the hud should manage and delete them and such.
What is the type HANGING the hud var?